### PR TITLE
Implemented deno chunk upload 

### PR DESCRIPTION
**Pull request by Meldiron**
Snippet:

```typescript
import * as sdk from "/Users/meldiron/Documents/GitHub/sdk-generator/examples/deno/mod.ts";

const client = new sdk.Client();
const storage = new sdk.Storage(client);

try {
    client
        .setEndpoint('http://localhost/v1') // Your API Endpoint
        .setProject('dev') // Your project ID
        .setKey('...') // Your secret API key
    ;

    const f = await storage.createFile('b', 'unique()', "dirik/big.mp4", [], [], (n) => {
        console.log("N", n);
    });
    console.log(f);
} catch(err) {
    console.error(err);
}
```

Output:
CleanShot 2022-01-24 at 14 01 00
CleanShot 2022-01-24 at 14 01 11
CleanShot 2022-01-24 at 14 01 15

### CODE REVIEW

File: templates/android/library/src/main/java/io/appwrite/Client.kt.twig [hunk 1]
```diff
@@ -327,8 +327,17 @@ class Client @JvmOverloads constructor(
                         .charStream()
                         .buffered()
                         .use(BufferedReader::readText)
+                        
                     val error = if (response.headers["content-type"]?.contains("application/json") == true) {
-                        body.fromJson()
+                        val map = gson.fromJson<Map<String, Any>>(
```
Comment (Meldiron) [inline templates/android/library/src/main/java/io/appwrite/Client.kt.twig:332]: I did not touch this file, this is probably difference between master and feat-perps branch. Syncing feat-prefs branch should solve this.

File: templates/android/library/src/main/java/io/appwrite/Client.kt.twig [hunk 2]
```diff
@@ -327,8 +327,17 @@ class Client @JvmOverloads constructor(
...
+                            body,
+                            object : TypeToken<Map<String, Any>>(){}.type
+                        )
+                        {{ spec.title | caseUcfirst }}Exception(
+                            map["message"] as? String ?: "", 
+                            (map["code"] as Number).toInt(), 
+                            body
+                        )
```
Comment (lohanidamodar) [inline templates/android/library/src/main/java/io/appwrite/Client.kt.twig:340]: @Meldiron why is this still here?

File: templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig [hunk 3]
```diff
@@ -286,8 +286,17 @@ class Client @JvmOverloads constructor(
...
-                        body.fromJson()
+                        val map = gson.fromJson<Map<String, Any>>(
+                            body,
+                            object : TypeToken<Map<String, Any>>(){}.type
+                        )
+                        {{ spec.title | caseUcfirst }}Exception(
+                            map["message"] as? String ?: "", 
+                            (map["code"] as Number).toInt(), 
+                            body
+                        )
```
Comment (lohanidamodar) [inline templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig:299]: This too?

**Review by lohanidamodar — COMMENTED** [review]
few concerns, not sure why kotlin change are here?

File: templates/android/library/src/main/java/io/appwrite/Client.kt.twig (hunk 1 above)
Comment (lohanidamodar) [reply templates/android/library/src/main/java/io/appwrite/Client.kt.twig:332]: I synced the feat-preps branch, however still see these changes

**Comment by lohanidamodar** [comment]
@Meldiron please revert the change in the kotlin files and I can merge this PR :pray

File: templates/deno/src/services/service.ts.twig [hunk 4]
```diff
@@ -69,8 +70,83 @@ export class {{ service.name | caseUcfirst }} extends Service {
...
+
+                    if (id) {
+                            headers['x-appwrite-id'] = id;
+                    }
+                    
+                    const totalBuffer = new Uint8Array(Client.CHUNK_SIZE);
+
+                    // 16kB chunks block limit: https://github.com/denoland/deno/discussions/9906
+                    for (let blockIndex = 0; blockIndex < 320; blockIndex++) {
+                            const buf = new Uint8Array(16384);
```
Comment (TorstenDittmann) [inline templates/deno/src/services/service.ts.twig:119]: lets make `16384` a const with the comment regarding the reference to why it is used 👍🏻

File: templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig [hunk 5]
```diff
@@ -286,8 +286,17 @@ class Client @JvmOverloads constructor(
...
                         .use(BufferedReader::readText)
+                        
                     val error = if (response.headers["content-type"]?.contains("application/json") == true) {
-                        body.fromJson()
+                        val map = gson.fromJson<Map<String, Any>>(
+                            body,
+                            object : TypeToken<Map<String, Any>>(){}.type
+                        )
+                        {{ spec.title | caseUcfirst }}Exception(
+                            map["message"] as? String ?: "",
```
Comment (TorstenDittmann) [inline templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig:296]: What does `as?` do?

Comment (lohanidamodar) [reply templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig:296]: This actually is a change in Kotlin and should be reverted.

File: templates/android/library/src/main/java/io/appwrite/Client.kt.twig (hunk 2 above)
Comment (Meldiron) [reply templates/android/library/src/main/java/io/appwrite/Client.kt.twig]: I manually reverted the changes

File: templates/android/library/src/main/java/io/appwrite/Client.kt.twig (hunk 1 above)
Comment (Meldiron) [reply templates/android/library/src/main/java/io/appwrite/Client.kt.twig]: I manually reverted the changes

File: templates/deno/src/services/service.ts.twig (hunk 4 above)
Comment (Meldiron) [reply templates/deno/src/services/service.ts.twig]: Stored the value in a const variable on Client, and started using it across the chunking system

**Comment by lohanidamodar** [comment]
@Meldiron Please check and undo the kotlin changes that are still there, I believe the reverted changes are now shown as change after you synced with 0.13 branch. Check and revert accordingly 🙏🏻

**Review by lohanidamodar — APPROVED** [review]

File: templates/deno/src/services/service.ts.twig [hunk 6]
```diff
@@ -69,8 +70,83 @@ export class {{ service.name | caseUcfirst }} extends Service {
...
+                            'content-range': 'bytes ' + start + '-' + end + '/' + size
+                    };
+
+                    if (id) {
+                            headers['x-appwrite-id'] = id;
+                    }
+                    
+                    const totalBuffer = new Uint8Array(Client.CHUNK_SIZE);
+
+                    // 16kB chunks block limit: https://github.com/denoland/deno/discussions/9906
```
Comment (TorstenDittmann) [inline templates/deno/src/services/service.ts.twig:117]: Can we move the comment to where we set `Client.CHUNK_SIZE`

Comment (lohanidamodar) [reply templates/deno/src/services/service.ts.twig:117]: Ah yes, it's there already, we need to remove this @Meldiron

Comment (Meldiron) [reply templates/deno/src/services/service.ts.twig:117]: Removed

**Pull request merged**